File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1-2/src/Commande.java
import java.util.Date;

import Voiture.Voiture;

public class Commande {
	private static int id = 0;
	private int idCommande;
	private Voiture[] voitures;
	private Date dateCommande;
	
	public Commande(Voiture[] voitures) {
		this.idCommande = id;
		Commande.id ++;
		this.voitures = voitures;
		this.dateCommande = new Date();
	}
	
	public Commande(Voiture voiture) {
		this.idCommande = id;
		Commande.id ++;
		this.voitures = new Voiture[1];
		this.voitures[0] = voiture;
		this.dateCommande = new Date();
	}
	
	public Commande() {
		this.idCommande = id;
		Commande.id ++;
		this.voitures = new Voiture[1];
		this.voitures[0] = null;
		this.dateCommande = new Date();
	}
	
	public Voiture[] getVoituresCommande() {
		return this.voitures;
	}
	
	public Date getDateCommande() {
		return this.dateCommande;
	}
	
	public void setVoituresCommande(Voiture[] voitures) {
		this.voitures = new Voiture[voitures.length];
		this.voitures = voitures;
	}
	
	public void setVoituresCommande(Voiture voiture) {
		this.voitures = new Voiture[1];
		this.voitures[0] = voiture;
	}
	
	public void addVoitureCommande(Voiture voiture) {
		Voiture[] addedVoitureTab = new Voiture[this.voitures.length +1];
		for (int i=0; i<this.voitures.length; i++) {
			/*if(this.voitures[i] != null)*/ addedVoitureTab[i] = this.voitures[i];
		}
		addedVoitureTab[addedVoitureTab.length -1] = voiture;
		this.voitures = addedVoitureTab;
	}
	
	public void addVoitureCommande(Voiture[] voitures) {
		Voiture[] addedVoitureTab = new Voiture[this.voitures.length + voitures.length+1];
		int i=0, j=0;
		for (i=0; i<this.voitures.length; i++) {
			addedVoitureTab[i] = this.voitures[i];
		}
		for (i=this.voitures.length; j<voitures.length; i++) {
			addedVoitureTab[i] = voitures[j];
			j++;
		}
		this.voitures = addedVoitureTab;
	}

	public String voituresTabToString() {
		String voituresTabString = "";
		if(this.voitures != null) {
			for(int i=0; i<this.voitures.length; i++) {
				if(this.voitures[i] != null) voituresTabString += this.voitures[i].toString();
			}
		}
		else if(this.voitures[0] == null && this.voitures[1] == null){
			voituresTabString = "aucune voiture";
		}
		
		return voituresTabString;
	}
	
	@Override
	public String toString() {
		return "Commande [idCommande=" + idCommande + " date=" + this.dateCommande.toString() + ", voitures=" + this.voituresTabToString() + "]";
	}
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Commande.php
<?php
class Commande{

    // attributs

    private $voitures = array();
    private static $id = 0;
    private $identifiantCommande;
    private $voitureTabString;

    // constructeur par initialisation
/*
    public function __construct(Voiture $voiture){
        $this->voiture = $voiture;
    }
*/
    public function __construct(){
    	$nb = func_num_args();
        $args = func_get_args();
        $commande = "commande$nb";
        if (method_exists(__CLASS__,$commande)){
            $this->$commande($args);
        }
    }

    public function commande0(){
        $this->identifiantCommande = $identifiantCommande;
        $identifiantCommande++;
        $this->voitures = new Voiture(1);
        $this->voitures[0] = null;
    }

    public function commande1($voiture){
        $this->identifiantCommande = self::$id;
        Commande::$id++;
        $this->voitures = new Voiture("",null,null,null,Couleur::NOIR,null);
        $this->voitures = $voiture;
    }

    public function commande3($voitures){
        $this->identifiantCommande = $id;
        Commande::$id++;
        $this->voitures[0] = $voiture;
    }
    
    // getters

    public function getVoituresCommande(){
        return $this->voitures;
    }

    // setters

    public function setVoituresCommandes($voitures){
        $this->voitures = new Voiture($voitures.length);
        $this->voitures = $voitures;
    }

    public function setVoituresCommande($voiture){
        $this->voitures = new Voiture(1);
        $this->voitures[0] = $voiture;
    }

    public function ajouterVoitureCommande($voiture){
        $ajouterVoitureTab = array();
        $ajouterVoitureTab = new Voiture($this->voitures->length + 1);
        for($i = 0; $i < $this->voitures->length;$i++){
            $ajouterVoitureTab[i] = $this->voitures[i];
        }
        $ajouterVoitureTab[$ajouterVoitureTab->length -1] = $voiture;
        $voitures = $ajouterVoitureTab;
    }

    public function ajouterVoitureCommande1(Voiture $voitures){
        $ajouterVoitureTab = array();
        $ajouterVoitureTab = new Voiture($this->voitures->length + $voitures->length+1);
        for($i = 0; $i < $this->voitures->length;$i++){
            $ajouterVoitureTab[i] = $this->voitures[i];     
        }
        for($i = $this->voitures->length;$j < $voitures->length;$i++){
            $ajouterVoitureTab[i] = $voitures[j];
            $j++;
        }
        $this->voitures = $ajouterVoitureTab;
    }

    public function voituresTabToString(){
        if($this->voitures != null){
            for($i = 0; $i < count($this->voitures);$i++){
                $this->voituresTabString = $this->voitures[$i]->__toString();
            }
        }else{
            if($this->voitures[0] == null && $this->voitures[1] == null){
                $voituresTabString = "aucune voiture";
            }
        }
        return $this->voituresTabString;
    }

    public function __toString(){
        return "Commande [identifiantCommande = " . $this->identifiantCommande . ", voitures = " . $this->voituresTabToString() . "]";
    }
}